lib/ui/Observers.js

```diff
@@ -247,146 +248,240 @@ export const ObserveInteraction = ({player, inactivityPeriod=3000, onSleep, onWa
+    case "c":
+      result = player.controls.ToggleTextTrack();
+      setRecentUserAction({action: result ? ACTIONS.SUBTITLES_ON : ACTIONS.SUBTITLES_OFF});
+      return true;
+    case "P":
+      player.controls.CollectionPlayPrevious();
+      setRecentUserAction({action: ACTIONS.PLAY_PREVIOUS});
+      return true;
+    case "N":
+      player.controls.CollectionPlayNext();
+      setRecentUserAction({action: ACTIONS.PLAY_NEXT});
+      return true;
+    case "Home":
+      player.controls.Seek({fraction: 0});
+      setRecentUserAction({action: ACTIONS.SEEK_BACK});
+      return true;
+    case "End":
+      player.controls.Seek({fraction: 1});
+      setRecentUserAction({action: ACTIONS.SEEK_FORWARD});
+      return true;
+    case "0":
+    case "1":
+    case "2":
+    case "3":
+    case "4":
+    case "5":
+    case "6":
+    case "7":
+    case "8":
+    case "9":
+      result = player.controls.Seek({fraction: parseFloat(event.key) * 0.1});
+      setRecentUserAction({action: result ? ACTIONS.SEEK_FORWARD : ACTIONS.SEEK_BACK});
+      return true;
+    default:
+      return false;
+  }
+};
+
+const NoArrowsKeydownHandler = ({player, event, setRecentUserAction, SeekHandler}) => {
+  switch (event.key) {
+    case "ArrowUp":
+    case "ArrowDown":
+    case "ArrowLeft":
+    case "ArrowRight":
+      return false;
+    default:
+      return DefaultKeydownHandler({player, event, setRecentUserAction, SeekHandler});
+  }
+};
+
+/**
+ * Get the first container we expect to find focusable elements in.
+ * This assumes that bottom-controls and info-box are mutually exclusive.
+ * Can return null if neither is visible.
+ */
+const getControlContainer = () => {
+  return document.querySelector("[class*='bottom-controls_'], [class*='info-box-container']");
+};
+
+/** Thanks, GPT */
+const getFocusableElements = () => {
+  return Array.from(
+    getControlContainer()
+      ?.querySelectorAll("button, [href], input, select, textarea, [tabindex]:not([tabindex=\"-1\"])")
+  ).filter((el) => !el.hasAttribute("disabled") && el.offsetParent !== null);
+};
+
+/**
+ * +1 for forward, -1 for backward.
+ * @param direction
+ */
+const moveFocus = (direction) => {
+  const focusableElements = getFocusableElements();
+  const activeIndex = focusableElements.indexOf(document.activeElement);
+  const nextIndex = (activeIndex + direction);
+  if(nextIndex >= 0 || nextIndex < focusableElements.length) {
+    focusableElements[nextIndex]?.focus();
+  }
+};
+
+const SpatialNavKeydownHandler = ({player, event, setRecentUserAction, SeekHandler}) => {
+  const currentFocus = document.activeElement;
+  const controlsRowHasFocus = getControlContainer()?.contains(currentFocus);
+  switch (event.key) {
+    case "ArrowUp":
+      if(player.controls.IsMenuVisible()) {
+        moveFocus(-1);
+        return true;
+      } else if(controlsRowHasFocus) {
+        document.getElementById(player.controls.GetControlId(PlayerControls.ID_SEEKBAR))?.focus();
```

Comment:
all instances of this pattern `player.controls.GetControlId(PlayerControls.ID_SEEKBAR)` can be replaced with `player.controls.GetPlayerControl(ElvPlayerControls.<control>)`, and PlayerControls shouldn't need to be imported anywhere except the player